daemon/fqdn.go

```diff
@@ -536,7 +533,7 @@ func extractDNSLookups(endpoints []*endpoint.Endpoint, CIDRStr, matchPatternStr
 	return lookups, nil
 }
 
-func deleteDNSLookups(endpoints []*endpoint.Endpoint, expireLookupsBefore time.Time, matchPatternStr string) (namesToRegen []string, err error) {
+func deleteDNSLookups(cache *fqdn.DNSCache, endpoints []*endpoint.Endpoint, expireLookupsBefore time.Time, matchPatternStr string) (namesToRegen []string, err error) {
```

Comment:
Alternatively, if you meant for this to be the poller cache, we should call it that. Then, we would need to do something different on line 552 since it now uses the wrong cache.